.gflows/libs/job_scan_code_net.lib.yml

```
@@ -0,0 +1,62 @@
+#@ load("common.lib.yml", "common")
+#@ load("configuration.lib.yml", "cfg")
+---
+#@ def generate_scan_code_net_job_steps(scan_code_net, sections):
+#@ test_result_artifacts = []
+
+#@ for unit_test_section in getattr(sections,"unit_test",[]):
+#@ test_result_artifacts.append("{} results".format(unit_test_section.name))
+#@ end 
+
+#@ if hasattr(sections,"integration_test") :
+  #@ for integration_test_definition in sections.integration_test:
+      #@ test_result_artifacts.append("{} results".format(integration_test_definition.name))
+  #@ end
+#@ end
+
+- name: Checkout repository
+  uses: actions/checkout@v2
+  with:
+    fetch-depth: 0
+- name: Scan
+  uses: NazmiAltun/dotnet-parallel-sonar-scan@v1.1.3
+  with:
+    sonar-token: #@ cfg.get_sonar_token(scan_code_net)
+    test-result-artifacts: #@ ",".join(test_result_artifacts)
+#@ if hasattr(scan_code_net.sonar,"coverage_solution_root_path") :
```

Please check for the existence of scan_code_net.sonar field, otherwise it is not optional and generation will fail. 
```suggestion
#@ if hasattr(getattr(scan_code_net,"sonar"),"coverage_solution_root_path")):
```

or a single root if for all fields 
```suggestion
#@ if hasattr(getattr(scan_code_net,"sonar")):
 .... many ifs here for every property
 end
```